projects/community-payback-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/AppointmentsService.kt

```diff
@@ -90,6 +92,56 @@ class AppointmentsService(
         )
     }
 
+    fun getSession(
+        projectCode: String,
+        date: LocalDate,
+        startTime: LocalTime,
+        endTime: LocalTime,
+        username: String
+    ): SessionResponse {
+        val project = unpaidWorkProjectRepository.getUpwProjectByCode(projectCode)
+        val appointments = unpaidWorkAppointmentRepository.getUpwAppointmentsByAppointmentDateAndStartTimeAndEndTime(
+            date, startTime, endTime
+        )
+
+        val appointmentSummaries = appointments.map {
+            val minutes = unpaidWorkAppointmentRepository.getUpwRequiredAndCompletedMinutes(it.upwDetailsId)
+
+            SessionResponseAppointmentSummary(
+                id = it.id,
+                case = AppointmentResponseCase(
+                    crn = it.person.crn,
+                    name = AppointmentResponseName(
+                        forename = it.person.forename,
+                        surname = it.person.surname,
+                        middleNames = it.person.secondName?.let { names -> listOf(names) } ?: emptyList()
+                    ),
+                    dateOfBirth = it.person.dateOfBirth,
+                    currentExclusion = it.person.currentExclusion,
+                    exclusionMessage = it.person.exclusionMessage,
+                    currentRestriction = it.person.currentRestriction,
+                    restrictionMessage = it.person.restrictionMessage,
+                ),
+                outcome = it.contact.contactOutcome?.let { contactOutcome ->
+                    CodeDescription(
+                        contactOutcome.code,
+                        contactOutcome.description,
+                    )
+                },
```

Comment:
Similarly here, it may be worth having a `ContactOutcome.toCodeDescription()` function?